app/services/access_process_service.rb

```diff
@@ -0,0 +1,75 @@
+class AccessProcessService
+  include RBAC::Permissions
+
+  def initialize(opts = {})
+    @app_name = opts[:app_name] || 'approval'
+    @prefix = opts[:role_prefix] || "#{@app_name}-group-"
+    @acls = RBAC::ACLS.new
+    @roles = RBAC::Roles.new(@prefix)
+    @policies = RBAC::Policies.new(@prefix)
+  end
+
+  # Add access control lists to the role for group
+  def add_acls(name, resource_id, permissions)
+    role = find_role(name)
+    raise Exceptions::RBACError("Role [#{name}] does not exist!") unless role
+
+    add_role_acls(role, resource_id, permissions)
+  end
+
+  def remove_acls(name, resource_id, permissions = [WORKFLOW_APPROVE_PERMISSION])
+    role = find_role(name)
+    raise Exceptions::RBACError("Role [#{name}] does not exist!") unless role
+
+    remove_acls_or_delete_role(role, resource_id, permissions)
+  end
+
+  def update_or_create_role(name, resource_id, permissions = [WORKFLOW_APPROVE_PERMISSION])
+    role = find_role(name)
+
+    role ? add_role_acls(role, resource_id, permissions) : create_role(name, resource_id, permissions)
+  end
+
+  def find_role(name)
+    @roles.find(name)
+  end
+
+  def add_resource_to_groups(resource_id, group_refs)
```

Comment:
It seems `add_resource_to_groups` and `remove_resource_from_groups` are the only two methods used outside. Add `permissions = [WORKFLOW_APPROVE_PERMISSION]` in the arg list and turn other public methods into private.